CRM/Core/BAO/ActionSchedule.php

```diff
@@ -600,9 +600,30 @@ protected static function pickFromEmail($actionSchedule) {
    *   List of error messages.
    */
   protected static function sendReminderEmail($tokenRow, $schedule, $toContactID) {
-    $toEmail = CRM_Contact_BAO_Contact::getPrimaryEmail($toContactID);
+    //CRM-18538 - check privacy settings
+    $checkPrivacy = CRM_Core_DAO::executeQuery("
+      SELECT c.id, c.is_opt_out, c.do_not_email, c.is_deceased, e.on_hold
+      FROM civicrm_contact c
+      LEFT JOIN civicrm_email e
+        ON c.id = e.contact_id
+        AND e.is_primary = 1
+      WHERE c.id = {$toContactID}
+    ");
+    while ($checkPrivacy->fetch()) {
+      if ($checkPrivacy->is_opt_out ||
+        $checkPrivacy->do_not_email ||
+        $checkPrivacy->is_deceased ||
+        $checkPrivacy->on_hold
+      ) {
+        $toEmail = '';
+      }
+      else {
+        $toEmail = CRM_Contact_BAO_Contact::getPrimaryEmail($toContactID);
+      }
+    }
+
     if (!$toEmail) {
-      return array("email_missing" => "Couldn't find recipient's email address.");
+      return array("email_missing" => "Couldn't find recipient's email address or privacy settings prevent sending.");
     }
```

Comment:
It's almost always better for performance to amalgamate two db queries, like here where there is one for  checkPrivacy one and another to retrieve the primary email. Wouldn't it be okay to do something like this instead:

$result = civicrm_api3('Contact', 'get', array(
  'sequential' => 1,
  'return' => array("email"),
  'do_not_email' => 0,
  'is_opt_out' => 0,
  'is_deceased' => 0,
  'on_hold' => 0,
  'id' => $toContactID,
  'email' => array('IS NOT NULL' => 1),
));